stable/phpmyadmin/templates/_helpers.tpl

```diff
@@ -0,0 +1,49 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "phpmyadmin.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "phpmyadmin.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "phpmyadmin.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "db.fullname" -}}
```

Comment:
All partials should be prefixed with the chart name, thus: `phpmyadmin.db.fullname`. Also, you should base it on the standard fullname template. Truncating as done here is probably not useful either because you'd strip off the db suffix. I'd just go with this:

`{{- template "phpmyadmin.fullname" . }}-{{ .Values.phpmyadmin.dbSuffix -}}`

But then I don't understand why this is needed or would work at all. In comments in `values.yaml` you mention this can be used if the DB is deployed as part of a release. In order for this to be possible there would need to be a dependency to a DB chart, but this chart doesn't have any dependencies.

Reply:
Hi @unguiculus, maybe I am doing something wrong here. The usecase I am trying to cater for is to include phpmyadmin is as a dependency itself. For example Wordpress. Wordpress chart has mariadb as a dependency. Now, lets say that we add phpmyadmin also.

Wordpress is going to have:
releasename-wordpress
releasename-mariadb
releasename-phpmyadmin

some how when I do helm install wordpress, I need to pass in the values.yaml a name for phpmyadmin to point to. If that name is fixed, I can just use phpmyadmin.dbHost , if it is not (like in this case) i can set phpmyadmin.dbSuffix = mariadb and would try to connect to releasename-mariadb. 

I there a better way to achieve this?

Reply:
I think this is fine, I agree with renaming this to `phpmyadmin.db.fullname` or something similar.

Reply:
This is a really cool feature, but I think the `suffix` parameter name is confusing/vague.
I would suggest renaming it something along the lines of `sqlServiceName`.

Reply:
the only thing is that is more like the chart name than the full service name, maybe sqlChartName?

Reply:
changed to db.chartName